CveXplore/main.py

```diff
@@ -144,7 +144,9 @@ def __init__(self, **kwargs):
         else:
             if self.datasource_type == "mongodb":
                 self._datasource_connection_details = {
-                    "host": f"{self.config.DATASOURCE_PROTOCOL}://{self.config.DATASOURCE_HOST}:{self.config.DATASOURCE_PORT}"
+                    "host": f"{self.config.DATASOURCE_PROTOCOL}://{self.config.DATASOURCE_HOST}:{self.config.DATASOURCE_PORT}" 
+                            if self.config.DATASOURCE_USER is None and self.config.DATASOURCE_PASSWORD is None
+                            else f"{self.config.DATASOURCE_PROTOCOL}://{self.config.DATASOURCE_USER}:{self.config.DATASOURCE_PASSWORD}@{self.config.DATASOURCE_HOST}:{self.config.DATASOURCE_PORT}" 
```

Comment:
The CI workflows, when accepted, will likely complain about Black formatting. This might fix it.

```suggestion
                    "host": f"{self.config.DATASOURCE_PROTOCOL}://{self.config.DATASOURCE_HOST}:{self.config.DATASOURCE_PORT}" 
                    if self.config.DATASOURCE_USER is None
                    and self.config.DATASOURCE_PASSWORD is None
                    else f"{self.config.DATASOURCE_PROTOCOL}://{self.config.DATASOURCE_USER}:{self.config.DATASOURCE_PASSWORD}@{self.config.DATASOURCE_HOST}:{self.config.DATASOURCE_PORT}"
```